Modules/_ctypes/_ctypes.c

```diff
@@ -459,20 +444,129 @@ static PyType_Spec structparam_spec = {
     .slots = structparam_slots,
 };
 
+/*
+  CType_Type - a base metaclass. Its instances (classes) have a StgInfo.
+  */
+
+static int
+CType_Type_traverse(PyObject *self, visitproc visit, void *arg)
+{
+    Py_VISIT(Py_TYPE(self));
+    return 0;
+}
+
+static void
+_ctype_clear_stginfo(StgInfo *info)
+{
+    assert(info);
+    Py_CLEAR(info->proto);
+    Py_CLEAR(info->argtypes);
+    Py_CLEAR(info->converters);
+    Py_CLEAR(info->restype);
+    Py_CLEAR(info->checker);
+}
+
+static int
+CType_Type_clear(PyObject *self)
+{
+    ctypes_state *st = GLOBAL_STATE();
+    if (st && st->PyCType_Type) {
+        StgInfo *info;
+        if (PyStgInfo_FromType(st, self, &info) < 0) {
+            return -1;
+        }
+        if (info) {
+            _ctype_clear_stginfo(info);
+        }
+    }
+    return 0;
+}
+
+static void
+CType_Type_dealloc(PyObject *self)
+{
+    ctypes_state *st = GLOBAL_STATE();
+
+    if (st && st->PyCType_Type) {
+        StgInfo *info;
+        if (PyStgInfo_FromType(st, self, &info) < 0) {
+            PyErr_WriteUnraisable(self);
+            PyErr_Clear();
```

Comment:
Hm, I might be misremembering, but I thought `PyErr_WriteUnraisable` also cleared the raised exception 🤔